tests/plugins/test_pandas_extensions.py

```diff
@@ -123,3 +126,23 @@ def test_pandas_html_writer(tmp_path: pathlib.Path) -> None:
     assert PandasHtmlWriter.applicable_types() == [pd.DataFrame]
     assert file_path.exists()
     assert metadata["path"] == file_path
+
+
+def test_pandas_gbq(df: pd.DataFrame) -> None:
+    mock_conn = mock.MagicMock()
+    mock_conn.run_query.return_value = df
+
+    with mock.patch("pandas_gbq.gbq.GbqConnector", return_value=mock_conn):
+        writer = PandasGbqWriter("project-id.dataset.table", project_id="test")
+        kwargs1 = writer._get_saving_kwargs()
+        writer.save_data(df)
+
+        reader = PandasGbqReader("SELECT foo FROM bar", max_results=1)
+        kwargs2 = reader._get_loading_kwargs()
+        df2, metadata = reader.load_data(pd.DataFrame)
```

Comment:
yeah this works.
What you'd usually do afterwards is check that the mock was called appropriately. E.g. assert that the right calls were made and they were passed in the right objects.

Reply:
I added an assert_called_once call in the unit test. Thanks for the feedback,  TIL 💡. Let me know if you think it makes sense to call any other assertion methods!